pkg/instance/otel.go

```diff
@@ -191,10 +192,10 @@ func (i *Instance) createOtelCollectorInstance(ctx context.Context) (*Instance,
 	if err := otelAgent.AddPortTCP(9090); err != nil {
 		return nil, ErrAddingOtelAgentPort.Wrap(err)
 	}
-	if err := otelAgent.SetCPU("100m"); err != nil {
+	if err := otelAgent.SetCPU(resource.MustParse("100m")); err != nil {
 		return nil, ErrSettingOtelAgentCPU.Wrap(err)
 	}
-	if err := otelAgent.SetMemory("100Mi", "200Mi"); err != nil {
+	if err := otelAgent.SetMemory(resource.MustParse("100Mi"), resource.MustParse("200Mi")); err != nil {
```

Comment:
Why aren't we using `memoryLimit` here? 

We should either use `memoryLimit` or define a new `memoryLimitOtel` with a comment explaining why it is different.

Reply:
@smuu do you know why is it different from executor?